### PR TITLE
Web color Example

### PR DESCRIPTION
**Pull request by arnau**
### Changes proposed in this pull request

Adds an example to introduce the basics of the command line interface.

### Guidance to review

The readme should be comprehensive and accurate.

### CODE REVIEW

File: examples/web-color/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+
+## Setup
+
+Ensure that you have Python +3.6 installed on your machine.
+
+Then, install the `registers` command line interface:
+
+```sh
+pip3 install --user "git+https://github.com/openregister/registers-cli#egg=registers"
```
Comment (gidsg) [inline examples/web-color/README.md]: This command didn't work for me due to conflicting installed dependencies. Perhaps you can include a virtualenv in the install to make the command more robust?

Comment (gidsg) [reply examples/web-color/README.md]: This command does not make `registers` command available for me, are you sure there is no need to add something to `PATH`?

File: examples/web-color/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+Now you can apply the patch to the register:
+
+```sh
+registers patch apply --rsf src/web-color.rsf patch_01.rsf
+```
+
+This last command _appends_ the content of the patch to the register so now
+we have some records to list. Let's do it in CSV:
```
Comment (gidsg) [inline examples/web-color/README.md]: I found this wording confusing. Maybe something like _So now let's preview the records as CSV_ would be clearer?

File: examples/web-color/README.md (hunk 1 above)
Comment (arnau) [reply examples/web-color/README.md]: If you haven't set `PATH` for user python installs pip mentions that, or at least it did for me. I don't think a tutorial should double on these but I'll think about it.

Regarding the virtualenv, I was desperately trying to avoid that. If you need to setup a virtualenv it should be your choice, not the tutorial's suggestion.

File: examples/web-color/README.md (hunk 2 above)
Comment (arnau) [reply examples/web-color/README.md]: Nice one, I'll change it

File: examples/web-color/README.md (hunk 1 above)
Comment (arnau) [reply examples/web-color/README.md]: I added a troubleshooting section. That said, both issues you mention are not related to this example nor to this project. Are very common situations in Python/Pip.

File: examples/web-color/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+## Structure of the example
+
+This example has:
+
+* The **register** for the web colors in `src/web-color.rsf`.
+* The **data** in TSV we will add to the register in `data/`.
+
+Note that the filenames could be anything, the content is what's relevant.
+
+
```
Comment (gidsg) [inline examples/web-color/README.md]: Can we state here or at the start of the next section that these commands should be run relative to `examples/web-color`?

e.g. 
> From your checkout directory `cd examples/web-color`?

I know the examples are self contained but from experience people often try to run commands from the checkout directory. e.g. this is how we document them previously in `deployment`.

Comment (arnau) [reply examples/web-color/README.md]: fair point 👍

File: examples/web-color/README.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+```sh
+register context show src/web-color.rsf
+```
+
+We could also get the record for green yellow, identified by `ADFF2F`, its
+hexadecimal code:
+
+```sh
+register record show --format json --rsf src/web-color.rsf ADFF2F
```
Comment (gidsg) [inline examples/web-color/README.md]: typo; should be `registers`

File: examples/web-color/README.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+This time, let's apply the patch at the same time as we create it:
+
+```sh
+registers patch create --apply --rsf src/web-color.rsf data/02.tsv
+```
+
+If we check the context again we'll see that there are 138 records in there:
+
+```sh
+register context show src/web-color.rsf
```
Comment (gidsg) [inline examples/web-color/README.md]: typo; should be `registers`

File: examples/web-color/README.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+```sh
+registers patch apply --rsf src/web-color.rsf patch_01.rsf
+```
+
+This last command _appends_ the content of the patch to the register so now
+we have some records to list. So now let's get the records as CSV:
+
+```sh
+register record list --format csv src/web-color.rsf
```
Comment (gidsg) [inline examples/web-color/README.md]: typo; should be `registers`

File: examples/web-color/README.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+## Publish the register
+
+The [Registers Specification V1](https://spec.openregister.org/v1) describes
+a REST API that any published register should comply with.
+
+In this example we will use Netlify to host our data and to add a bit of route
+handling to align with the reference implementation (ORJ) custom behaviour.
+
+```sh
+registers build --target netlify src/03_patch02.rsf
```
Comment (gidsg) [inline examples/web-color/README.md]: When I ran this I got
```
Error: Invalid value for "RSF_FILE": Path "src/03_patch02.rsf" does not exist.
```
I think it should be `registers build --target netlify src/web-color.rsf`?

File: examples/web-color/README.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+```
+
+This command creates all the necessary files for a static deployment. Given
+that the register identifier is `web-color` you should expect a
+`build/web-color` directory.
+
```
Comment (gidsg) [inline examples/web-color/README.md]: I know this would be a separate PR but I think it would be helpful if the build command printed the absolute path to the created folder. e.g. Site successfully created at: 
```/Users/gideongoldberg/git/registers-cli/examples/web-color/build/web-color```
Else the build folder is implicit knowledge. 
We do something similar in the Ruby spike: 
https://github.com/openregister/static-register-generator/blob/f3d1456fdf6561bd6299dc75a5adf3d0c46e2027/generate_register.thor#L35
